src/createServer.js

```diff
@@ -1,10 +1,38 @@
-/* eslint-disable no-console */
+
 'use strict';
 
-function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
-}
+const http = require('http');
+
+const createServer = () => {
+  const server = http.createServer((req, res) => {
+    const normalizeURL = req.url.replace(new RegExp('/+'), '/');
+    const path = new URL(normalizeURL, 'http://localhost');
+    const parts = path.pathname.split('/').filter((e) => e.length > 0);
+
+    const query = {};
+
+    path.searchParams.forEach((value, key) => {
+      if (query[key] === undefined) {
+        query[key] = value;
+      } else if (Array.isArray(query[key])) {
+        query[key].push(value);
+      } else {
+        query[key] = [query[key], value];
+      }
+    });
```

Comment:
You can try ( optional) rewrite using reduce. 
Also, type condition is not best idea ( it`s about undefined)